custom_components/ha_kia_hyundai/select.py

```diff
@@ -0,0 +1,142 @@
+"""Select entity for seats."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Final
+
+from custom_components.ha_kia_hyundai import VehicleCoordinator
+from custom_components.ha_kia_hyundai.vehicle_coordinator_base_entity import (
+    VehicleCoordinatorBaseEntity,
+)
+
+from homeassistant.components.select import SelectEntity, SelectEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_UNKNOWN, STATE_UNAVAILABLE
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.restore_state import RestoreEntity
+
+from .const import CONF_VEHICLE_ID, DOMAIN, SEAT_STATUS, STR_TO_NUMBER
+
+OFF = ["Off"]
+HEAT_OPTIONS = {
+    3: ["High Heat", "Medium Heat", "Low Heat"],
+    2: ["High Heat", "Low Heat"],
+}
+COOL_OPTIONS = {
+    3: ["High Cool", "Medium Cool", "Low Cool"],
+    2: ["High Cool", "Low Cool"],
+}
+
+HEAT_TYPE = "heatVentType"
+STEPS = "heatVentStep"
+
+
+@dataclass(frozen=True, kw_only=True)
+class KiaSelectEntityDescription(SelectEntityDescription):
+    """Class for Kia select entities."""
+
+    exists_fn: Callable[[VehicleCoordinatorBaseEntity], bool] = lambda _: True
+    value_fn: Callable[[VehicleCoordinatorBaseEntity], str | None]
+    options_fn: Callable[[VehicleCoordinatorBaseEntity], int | None]
+    icon = "mdi:car-seat"
+
+
+SEAT_SELECTIONS: Final[tuple[KiaSelectEntityDescription, ...]] = (
+    KiaSelectEntityDescription(
+        key="desired_driver_seat_comfort",
+        name="Seat-Driver with Climate",
+        exists_fn=lambda seat: bool(seat.front_seat_options[HEAT_TYPE]),
```

Comment:
instead of seat these parameters should be coordinator to match what is passed in